File: code/SlidingWindow/SmallestSubarrayGivenSum/SmallestSubarrayGivenSum.js
const smallestSubarrayGivenSum = function(s, array){
    // handling edge case if there is an empty array
    if(!array.length) return 0;

    // instantiating all the variables needed 
    let right = 0;
    let left = 0;
    // using the global property of Infinity
    let minimumSize = Infinity;
    let currentSum = 0;

    while (right != array.length){
        currentSum += array[right];
        while (currentSum >= s){
            minimumSize = Math.min(minimumSize, right - left + 1);
            currentSum -= array[left];
            left ++;
        }
        right ++;
    }
    return minimumSize;
}

File: code/SlidingWindow/SmallestSubarrayGivenSum/SmallestSubarrayGivenSum.java
package code.SlidingWindow.SmallestSubarrayGivenSum;

public class SmallestSubarrayGivenSum {
    public int smallSubArrayLength(int s, int[] array){
        int minimumLength = Integer.MAX_VALUE;
        int right = 0;
        int currentSum = 0;

        if(array.length == 0) return 0;

        // did js with while and java with for loop to change it up
        
        for(int left = 0; left < array.length; left++){
            currentSum += array[left];
            while(currentSum >= s){
                minimumLength = Math.min(minimumLength, left - right + 1);
                currentSum -= array[left];
                left++;
            }
        }
        return minimumLength;
    }
}